sdk/search/azure-search-documents/azure/search/documents/indexes/aio/_search_indexer_client.py

```diff
@@ -278,6 +278,31 @@ async def reset_indexer(self, name, **kwargs):
         kwargs["headers"] = self._merge_client_headers(kwargs.get("headers"))
         await self._client.indexers.reset(name, **kwargs)
 
+    @distributed_trace_async
+    async def reset_documents(self, indexer, keys_or_ids, **kwargs):
+        # type: (Union[str, SearchIndexer], DocumentKeysOrIds, **Any) -> None
+        """Resets specific documents in the datasource to be selectively re-ingested by the indexer.
+
+        :param indexer: The indexer to reset documents for.
+        :type indexer: str or ~azure.search.documents.indexes.models.SearchIndexer
+        :param keys_or_ids:
+        :type keys_or_ids: ~azure.search.documents.indexes.models.DocumentKeysOrIds
+        :return: None, or the result of cls(response)
+        :keyword overwrite: If false, keys or ids will be appended to existing ones. If true, only the
+         keys or ids in this payload will be queued to be re-ingested. The default is false.
+        :paramtype overwrite: bool
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        kwargs["headers"] = self._merge_client_headers(kwargs.get("headers"))
+        kwargs["keys_or_ids"] = keys_or_ids
+        try:
+            name = indexer.name
+        except AttributeError:
+            name = indexer
+        result = await self._client.indexers.reset_docs(name, **kwargs)
+        return
```

Comment:
Do we really need it? (doesn't mypy complain?)

Reply:
Sorry, that should return result.